packages/core/src/message-provider.ts

```diff
@@ -91,11 +91,17 @@ export function createMessageProvider({
         });
 
         for (const { _id, encryptedMessage } of encryptedMessages) {
-          await wallet.addDocument({
-            id: _id,
-            type: 'DIDCommMessage',
-            encryptedMessage,
-          });
+          try {
+            await wallet.addDocument({
```

Comment:
at what point are these messages removed?

Reply:
there is a markMessageAsRead function exposed on this file. The wallet will listen for a `wallet.eventManager.emit('didcomm-message-decrypted', decryptedMessage);` and will call markMessageAsRead after processing it